### PR TITLE
S2i example and update submodule

### PR DESCRIPTION
**Pull request by jscotka**

### CODE REVIEW

File: examples/s2i-python-sample/Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+TESTS=$(shell ls *.py)
+
+check:
+	# run modulelint explicitly in example
+	#mtf -l
+	mtf $(TESTS)
+
+all: check
+
+.PHONY: all
```
Comment (TomasTomecek) [inline examples/s2i-python-sample/Makefile]: +check

File: examples/s2i-python-sample/simpleTest.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        self.assertIn("Hello from gunicorn WSGI application!", output.stdout)
+
+    def test_basic(self):
+        self.start()
+        self.life_check()
+
+    def test_another_port(self):
+        self.backend.info["start"]="docker run -p 9999:8080"
+        self.start()
+        self.life_check(port="9999")
```
Comment (TomasTomecek) [inline examples/s2i-python-sample/simpleTest.py]: one more test could be to check `s2i usage` whether it's present

Comment (jscotka) [reply examples/s2i-python-sample/simpleTest.py]: test added as separate class in this file

File: examples/s2i-python-sample/usageTest.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Jan Scotka <jscotka@redhat.com>
+#
+
+from moduleframework import module_framework
+
+class usageTest(module_framework.ContainerAvocadoTest):
```
Comment (TomasTomecek) [inline examples/s2i-python-sample/usageTest.py]: should be `UsageTest`

File: examples/s2i-python-sample/Makefile [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+check-python-3.5:
+	MODULE=docker-python-35 mtf $(TESTS)
+
+check-usage-minimal:
+	CONFIG=$(MINIMAL_CONFIG) URL=docker=centos/python-35-centos7 MODULE=docker mtf $(TESTS_GENERAL)
+	CONFIG=$(MINIMAL_CONFIG) URL=docker=centos/python-27-centos7 MODULE=docker mtf $(TESTS_GENERAL)
+
+all: check
+
+.PHONY: all
```
Comment (TomasTomecek) [inline examples/s2i-python-sample/Makefile]: more missing targets for `.PHONY`

**Pull request merged**